pkgs/by-name/am/amd-ucodegen/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  callPackage,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "amd-ucodegen";
+  version = "0-unstable-2017-06-07";
+
+  src = fetchFromGitHub {
+    owner = "AndyLavr";
+    repo = "amd-ucodegen";
+    rev = "0d34b54e396ef300d0364817e763d2c7d1ffff02";
+    hash = "sha256-pgmxzd8tLqdQ8Kmmhl05C5tMlCByosSrwx2QpBu3UB0=";
+  };
+
+  strictDeps = true;
+
+  patches = [ ./validate-family-not-id.patch ];
```

Comment:
Can you add a comment to this patch, per guidelines? Is this from a PR, if so, can we link to it? If not, where did the patch come from?

Reply:
There is a description inside the patch file itself (is it better to add it as a comment to `package.nix`?)

> Extract get_family function and validate processor family instead of processor ID.
> Source: https://github.com/AndyLavr/amd-ucodegen/compare/master...dobo90:amd-ucodegen:master

The source is from a fork, I thought it was better to use the original upstream and then patch it ourselves, but we can directly use the fork if you think it's better.

Reply:
Oh, upstream is 7 years old... hmm then we should probably use the maintained fork.
But yes, normally per https://github.com/NixOS/nixpkgs/tree/master/pkgs#patches we should add a comment in the nix file itself regarding the patch, and fetch it if possible.

Reply:
The fork is directly used for the [amd-zen-ucode-platomav](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=amd-zen-ucode-platomav) package in the Arch AUR (by the package maintainer themselves). 

I don't really know if the fork is "maintained" or simply created ad-hoc.

Reply:
Actually, we could directly fetch https://github.com/AndyLavr/amd-ucodegen/compare/0d34b54e396ef300d0364817e763d2c7d1ffff02...dobo90:amd-ucodegen:7a3c51e821df96910ecb05b22f3e4866b4fb85b2.patch  if we don't want to use the fork.

(I just learnt about adding `.patch` at the end of a GitHub link).

Reply:
Implemented remote patch fetching, but I can directly use the fork if needed.